st2auth/st2auth/sso/__init__.py

```diff
@@ -0,0 +1,76 @@
+# Copyright 2019 Extreme Networks, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import
+
+import json
+import six
+import traceback
+
+from oslo_config import cfg
+
+from st2common import log as logging
+
+from st2common.util import driver_loader
+
+
+__all__ = [
+    'get_available_backends',
+    'get_backend_instance',
+    'get_sso_backend'
+]
+
+LOG = logging.getLogger(__name__)
+
+BACKENDS_NAMESPACE = 'st2auth.sso.backends'
+
+
+def get_available_backends():
+    return driver_loader.get_available_backends(namespace=BACKENDS_NAMESPACE)
+
+
+def get_backend_instance(name):
+    sso_backend_cls = driver_loader.get_backend_driver(namespace=BACKENDS_NAMESPACE, name=name)
+
+    kwargs = {}
+    sso_backend_kwargs = cfg.CONF.auth.sso_backend_kwargs
+
+    if sso_backend_kwargs:
```

Comment:
We use this pattern in multiple places (auth backend, sso backend, rbac backend) so it might make sense to refactor it in a new utility method.

Reply:
There is some similarity in `get_backend_instance` between the auth backend and sso backend.  But the rbac backend's `get_backend_instance` is different.  auth and sso version returns an instance and rbac returns a class.  I rather tackle this separately.